qa/L0_s3_local/test.sh

```diff
@@ -222,6 +222,75 @@ wait $SERVER_PID
 awslocal $ENDPOINT_FLAG s3 rm s3://demo-bucket1.0 --recursive --include "*" && \
     awslocal $ENDPOINT_FLAG s3 rb s3://demo-bucket1.0
 
+# Test for multiple model stores using S3 cloud repositories
+BACKENDS1="graphdef libtorch"
+BACKENDS2="onnx plan savedmodel"
+BACKENDS="$BACKENDS1 $BACKENDS2"
+
+rm -rf models1 && mkdir models1
+for BACKEND in $BACKENDS1; do
+    cp -r $DATADIR/${BACKEND}_float32_float32_float32 models1/.
+    # Remove version policy from config.pbtxt
+    sed -i '/^version_policy/d' models1/${BACKEND}_float32_float32_float32/config.pbtxt
+done
+
+rm -rf models2 && mkdir models2
+for BACKEND in $BACKENDS2; do
+    cp -r $DATADIR/${BACKEND}_float32_float32_float32 models2/.
+    # Remove version policy from config.pbtxt
+    sed -i '/^version_policy/d' models2/${BACKEND}_float32_float32_float32/config.pbtxt
+done
+
+BUCKET_NAME="demo-bucket"
+for BUCKET_SUFFIX in 1 2; do
+    # Cleanup bucket if exists
+    awslocal $ENDPOINT_FLAG s3 rm s3://$BUCKET_NAME$BUCKET_SUFFIX --recursive --include "*" && \
+        awslocal $ENDPOINT_FLAG s3 rb s3://$BUCKET_NAME$BUCKET_SUFFIX || true
+
+    # Create and add data to bucket
+    awslocal $ENDPOINT_FLAG s3 mb s3://$BUCKET_NAME$BUCKET_SUFFIX && \
+        awslocal $ENDPOINT_FLAG s3 sync models$BUCKET_SUFFIX s3://$BUCKET_NAME$BUCKET_SUFFIX
+done
+
+SERVER_ARGS="--model-repository=s3://localhost:4572/demo-bucket1 --model-repository=s3://localhost:4572/demo-bucket2 --model-control-mode=explicit --log-verbose=1"
+SERVER_LOG="./inference_server.multi.log"
+
+run_server
+if [ "$SERVER_PID" == "0" ]; then
+    echo -e "\n***\n*** Failed to start $SERVER\n***"
+    cat $SERVER_LOG
+    # Kill minio server
+    kill $MINIO_PID
+    wait $MINIO_PID
+    exit 1
+fi
+
+set +e
+for BACKEND in $BACKENDS; do
+    code=`curl -s -w %{http_code} -X POST localhost:8000/v2/repository/models/${BACKEND}_float32_float32_float32/load`
+    if [ "$code" != "200" ]; then
+        echo -e "\n***\n*** Test Failed\n***"
+        RET=1
+    fi
+
+    $PERF_CLIENT -m ${BACKEND}_float32_float32_float32 -p 3000 -t 1 >$CLIENT_LOG 2>&1
```

Comment:
> Are you suggesting that the outputs from the perf_analyzer could be different if the model is loaded from S3?


The only reason to add that validation here would be if you believe there is a possible scenario for the same. If we add that here I could like to do it across all cloud tests together and in a different ticket. However I cannot reason why it should be needed.

Reply:
> If we add that here I could like to do it across all cloud tests together and in a different ticket. However I cannot reason why it should be needed.

Why do you think it is not needed? I, on the other hand believe we must enable output_validation mode wherever possible to improve our test coverage.

Reply:
Sure. 

> Why do you think it is not needed?

I agree that we should be adding output validation to our tests. But rather it should be independent of S3 or any file system type. I do not see why adding it specifically to S3 instead of local filesystem helps, especially for a simple addsub model. 

If the model loads and runs successfully do we not believe that the accuracy will always be independent of the file system that was used to pass it to Triton?

Reply:
> Why you use perf analyzer for this test? We already have the test utils for checking correctness with QA models, and the test here doesn't seems to be benchmarking which may make sense to use perf analyzer.

There is no reason to use perf analyzer specifically. We should technically replace it with a simple infer test.

Reply:
Should use the test util to test inference then, using perf_analyzer typically implies benchmarking / maintaining certain load which is not the intention in this case..

Reply:
I agree with Guan. It will be better to use test util infer if you don't want to test the system under load. It will take a longer time to run perf_analyzer.

Reply:
Filing a ticket to make this change across all cloud storage tests.